### PR TITLE
Refactor pipeline parse datetime/time of MySQL binlog

### PR DESCRIPTION
**Pull request by azexcy**
Now, pipeline increment tasks of always return string of MySQL time type, it's different type from inventory tasks.

Changes proposed in this pull request:
  - Refactor pipeline parse datetime/time of MySQL binlog

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLDatetime2BinlogProtocolValue.java [hunk 1]
```diff
@@ -46,15 +48,16 @@ private long readDatetimeV2FromPayload(final MySQLPacketPayload payload) {
...
-        return String.format("%02d:%02d:%02d", time >> 12, (time >> 6) % (1 << 6), time % (1 << 6));
+        int year = (int) (yearAndMonth / 13);
+        int month = (int) (yearAndMonth % 13);
+        int day = (int) (date % (1 << 5));
+        long time = datetimeWithoutSign % (1 << 17);
+        int hour = (int) (time >> 12);
+        int minute = (int) ((time >> 6) % (1 << 6));
+        int second = (int) (time % (1 << 6));
+        MySQLFractionalSeconds fractionalSeconds = new MySQLFractionalSeconds(columnDef.getColumnMeta(), payload);
+        return Timestamp.valueOf(LocalDateTime.of(year, month, day, hour, minute, second, fractionalSeconds.getNanos()));
```
Comment (azexcy) [inline db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLDatetime2BinlogProtocolValue.java:61]: eg.  value is`2023-03-14 19:30:16.770925`
it's the above year, month, day, hour, minute and second, the fractional seconds is `.770925`

File: db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLFractionalSeconds.java [hunk 2]
```diff
@@ -26,41 +27,29 @@
...
     
     private final int fractionalSecondsPrecision;
     
     public MySQLFractionalSeconds(final int columnMeta, final MySQLPacketPayload payload) {
         fractionalSecondsPrecision = columnMeta;
-        fraction = readFraction(payload);
+        nanos = readFraction(payload);
     }
     
     private int readFraction(final MySQLPacketPayload payload) {
```
Comment (azexcy) [inline db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLFractionalSeconds.java:40]: eg. the fractional seconds is `.77` means the precision is 2, equals 0.77 seconds, to nanos is `0.77 * 1000 * 1000 * 1000`, paylod.readInt1() will return 77
the fractional seconds is `.773` means the precision is 3, equals 0.773 seconds, to nanos is `0.773 * 1000 * 1000`, paylod.readUnsignedShort() will return 773

**Comment by azexcy** [comment]
test example :
1. update t_datetime = '2023-03-15 11:03:37.123457 
image

2. update t_datetime = '2023-03-15 11:03:37.023000'
image

3. update t_datetime = '2023-03-15 11:03:37.010000'
image

File: db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLFractionalSeconds.java [hunk 3]
```diff
@@ -26,41 +27,29 @@
...
-        if (0 == fractionalSecondsPrecision) {
-            return "";
-        }
-        StringBuilder result = new StringBuilder(Integer.toString(fraction));
-        for (int i = result.length(); i < fractionalSecondsPrecision; i++) {
-            result.append("0");
-        }
-        result.setLength(fractionalSecondsPrecision);
-        return "." + result;
-    }
```
Comment (azexcy) [inline db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLFractionalSeconds.java:65]: There have problem when the fractional seconds just like `025000`

1. `payload.getByteBuf().readUnsignedMedium();` return `25000`
2. toString will append 0 at end, value = `250000`

so the result is wrong

File: db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLFractionalSeconds.java (hunk 2 above)
Comment (sandynz) [inline db-protocol/mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/packet/binlog/row/column/value/time/MySQLFractionalSeconds.java:40]: `readFraction` could be `convertFractionalSecondsToNanos`

**Review by sandynz — CHANGES_REQUESTED** [review]

**Review by sandynz — APPROVED** [review]

**Pull request merged**